### PR TITLE
feat(sumologicexporter): remove setting source headers

### LINKED ISSUE #255: [sourceprocessor,sumologicexporter] Remove source templates
Source templates are implemented in two places:
 - sourceprocessor (with addidional support for pod_name and possibility to be replaced by pod annotations)
 - sumologicexporter
 
 This is against DRY and to resolve it I propose the following:
 
 - remove source templates from sourceprocessor
 - add ability to define source templates using attribute in sumologicexporter with fallback to current approach
 
 With such solution, sourceprocessor will be able to prepare template based on pod annotations and exporter will take care of rest.

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
Resolves #255 
Ref: #685 

This PR removes the functionality of setting source headers in the `sumologic` exporter, which has been deprecated some time ago.

### CODE REVIEW

File: pkg/exporter/sumologicexporter/config.go [hunk 1]
```diff
@@ -144,6 +132,12 @@ Please consult the changelog at https://github.com/SumoLogic/sumologic-otel-coll
 		)
 	}
 
+	if cfg.SourceCategory != "" || cfg.SourceHost != "" || cfg.SourceName != "" {
+		return fmt.Errorf(`*Deprecation warning*: setting source headers is not supported anymore.
+Please consult the changelog at https://github.com/SumoLogic/sumologic-otel-collector/releases/tag/v0.60.0-sumo-0`,
+		)
+	}
+
```
Comment (sumo-drosiek) [inline pkg/exporter/sumologicexporter/config.go:140]: Do we need it?

Comment (aboguszewski-sumo) [reply pkg/exporter/sumologicexporter/config.go:140]: I'm not sure if we really need it. I found some time ago that such warnings have been printed so I added it. 
It was also added for attribute translation and telegraf metrics translation.

Comment (sumo-drosiek) [inline pkg/exporter/sumologicexporter/config.go:140]: Do we need it?

Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/config.go:140]: We added that because it was deprecated, now it's going to be removed :)

Comment (aboguszewski-sumo) [reply pkg/exporter/sumologicexporter/config.go:140]: The first such warning was this: https://github.com/SumoLogic/sumologic-otel-collector/blob/19c766e288aacae3e1a757a74a87ffc6a824af8b/pkg/exporter/sumologicexporter/config.go#L128-L134

Comment (swiatekm-sumo) [reply pkg/exporter/sumologicexporter/config.go:140]: I think this is ok, but it probably shouldn't say `deprecation warning`.

Comment (aboguszewski-sumo) [reply pkg/exporter/sumologicexporter/config.go:140]: @sumo-drosiek @swiatekm-sumo what's the conclusion then? I'm not against any of the options, but if we don't want it here, we should probably also remove these warnings for other deleted options.

Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/config.go:140]: @swiatekm-sumo we can do that

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**